### PR TITLE
Refactor `getDataFromColumnPath` logic 

### PR DESCRIPTION
**Pull request by julieg18**
* simplify code and reduce duplication
* ~fix bug with long integers~

Followup to comments in #2956

### CODE REVIEW

File: extension/src/experiments/model/quickPick.test.ts [hunk 1]
```diff
@@ -147,7 +147,7 @@ describe('pickExperiments', () => {
           description: '[exp-456]',
           detail: `Created:${formatDate(
             mockedExperiments[1].Created as string
-          )}, split:2.2000e+9, data/data.xml:22a1a29`,
```
Comment (julieg18) [inline extension/src/experiments/model/quickPick.test.ts:150]: Spotted this bug that was happening with long integers while working on the refactoring.

Comment (mattseddon) [reply extension/src/experiments/model/quickPick.test.ts:150]: This was actually intended (see https://github.com/iterative/vscode-dvc/pull/2813#discussion_r1029869455). 

There is a long tail of edge cases here. For example, why would we treat `2200043556.1` differently? What if we had `10000000000000000`? The JS numbers rabbit hole is deep.

File: extension/src/experiments/model/util.ts [hunk 2]
```diff
@@ -8,8 +8,10 @@ type Value = undefined | null | [] | string | number
 const isDate = (value: Value): boolean =>
   !!(typeof value === 'string' && Date.parse(value))
 
-const isLongNumber = (value: Value): boolean =>
-  typeof value === 'number' && value.toString().length > 7
+const isLongFloatNumber = (value: Value): boolean =>
+  typeof value === 'number' &&
+  !Number.isInteger(value as number) &&
+  value.toString().length > 7
```
Comment (mattseddon) [inline extension/src/experiments/model/util.ts:14]: personally, I would revert this change.

**Review by sroy3 — APPROVED** [review]

File: extension/src/experiments/model/quickPick.test.ts (hunk 1 above)
Comment (julieg18) [reply extension/src/experiments/model/quickPick.test.ts:150]: Oh, whoops! Must have missed that comment! The table has the full number so I assumed it was a bug. Will revert this change :)

Comment (mattseddon) [reply extension/src/experiments/model/quickPick.test.ts]: Do you think we should update the webview?

Comment (julieg18) [reply extension/src/experiments/model/quickPick.test.ts]: > Do you think we should update the webview?

Sure! Apologies if that's what you wanted originally, I misunderstood. When you say webview, are you referring to just the tooltips or the table as well? I'll open a pr that does both for now.

Comment (mattseddon) [reply extension/src/experiments/model/quickPick.test.ts]: I didn't even think about that until you replied. No need to apologise. It was probably me that missed something back in the original PR.
